### PR TITLE
Refactor Video, Transcript, and Caption Entities for Extensibility

### PR DESCRIPTION
**Pull request by movchan74**
**Summary:**
This update restructures our data models for videos, transcripts, and captions to improve extensibility and prepare for future app-specific implementations. It introduces a clear separation between base and extended entities, establishes polymorphic relationships, and updates related code and tests.

**Key Changes:**
1. **Entity Splitting**: 
   - Split VideoEntity into VideoEntity (base) and ExtendedVideoEntity (extended with duration, status, captions, transcript).
   - Split TranscriptEntity into TranscriptEntity (base) and ExtendedVideoTranscriptEntity (with media ID to ExtendedVideoEntity).
   - Split CaptionEntity into CaptionEntity (base) and ExtendedVideoCaptionEntity (with media ID to ExtendedVideoEntity).
   - Note: Extended entities are app-specific and will be moved to the app repo later.

2. **Polymorphic Relationships**: Established polymorphic relationships for extensibility:
   - MediaEntity <- VideoEntity <- ExtendedVideoEntity
   - TranscriptEntity <- ExtendedVideoTranscriptEntity
   - CaptionEntity <- ExtendedVideoCaptionEntity

3. **Code Reorganization**: Migrated functions from aana/storage/services to corresponding repositories.

4. **Testing Improvements**: Removed unit tests for DB operations and replaced them with integration tests for more comprehensive testing.

5. **Database Updates**: Replaced migrations to support the new database structure.

I wanted to add support for PostgreSQL for the tests but it's already huge PR.  Instead, I've added the issue [Add Support for PostgreSQL to Tests ](https://github.com/mobiusml/aana_sdk/issues/139). 

I also tried to optimize the MediaEntity creation in MediaRepository but ran into some issues. I've added a ticket for that: [Improve Media Creation in Media Repository](https://github.com/mobiusml/aana_sdk/issues/140).

This PR is a preparation for https://github.com/mobiusml/aana_sdk/issues/129.

### CODE REVIEW

**Review by evanderiel — APPROVED** [review]
LGTM

File: aana/alembic/versions/9cf76699210f_init.py [hunk 1]
```diff
@@ -29,19 +70,20 @@ def upgrade() -> None:
             comment="Unique identifier for the media",
         ),
         sa.Column(
-            "media_type", sa.String(), nullable=True, comment="The type of media"
+            "media_type", sa.String(), nullable=False, comment="The type of media"
```
Comment (HRashidi) [inline aana/alembic/versions/9cf76699210f_init.py:73]: Msst of the nullable attrs are changed to the false, It is okey to use base model (not extended version) to add item without the captions and other stuff?

**Review by HRashidi — APPROVED** [review]
Good jobs.
Just a questions.

ps:
From the Appu's note can we just move the db initiation as part of projects? It can be a little duplicate the code, but it clearly separate the db (also its type, mongo, postgres) from the core.

**Comment by movchan74** [comment]
@HRashidi 

> From the Appu's note can we just move the db initiation as part of projects? It can be a little duplicate the code, but it clearly separate the db (also its type, mongo, postgres) from the core.

Could you please expand on it? We still need DB in the SDK right now, at least for the task queue. It is possible to move all database operations to the projects itself but it wouldn't be a bit of code. We can also move most DB stuff into the app template but I'm not sure if it makes it easier. Also, it's not really what Appu wants if I understand him correctly. He doesn't want to have SQL DB at all and to use Qdrant for everything.

**Comment by HRashidi** [comment]
> @HRashidi
> 
> > From the Appu's note can we just move the db initiation as part of projects? It can be a little duplicate the code, but it clearly separate the db (also its type, mongo, postgres) from the core.
> 
> Could you please expand on it? We still need DB in the SDK right now, at least for the task queue. It is possible to move all database operations to the projects itself but it wouldn't be a bit of code. We can also move most DB stuff into the app template but I'm not sure if it makes it easier. Also, it's not really what Appu wants if I understand him correctly. He doesn't want to have SQL DB at all and to use Qdrant for everything.

Yea. I meant to have a abstraction to connect the db into the projects instead of the aana itself. So even if the user just want the Qdrant, he can implement all the stuff with it (moving the migration version and db classes into the each project). but I forgot about the task queue. Seems like we need the sql anyway.

**Pull request merged**